pkg/crypto/pedersen/points.go

```diff
@@ -2049,5 +2048,76 @@ func init() {
 		y, _ := new(big.Int).SetString(p[1], 16)
 		points[i] = point{x, y}
 	}
-	curve = weierstrass.Stark()
+	P, _ = new(big.Int).SetString("800000000000011000000000000000000000000000000000000000000000001", 16)
+	A = big.NewInt(1)
+}
+
+// Add returns the sum of (x1, y1) and (x2, y2) on an
+// elliptic curve mod P.
+// See https://github.com/starkware-libs/cairo-lang/blob/2abd303e1808612b724bc1412b2b5babd04bb4e7/src/starkware/python/math_utils.py#L147-L164
+func (p *point) Add(p2 point) {
+	if p.x.Sign() == 0 && p.y.Sign() == 0 {
+		p.x.Set(p2.x)
+		p.y.Set(p2.y)
+		return
+	}
+
+	if p2.x.Sign() == 0 && p2.y.Sign() == 0 {
+		return
+	}
+
+	x1, y1 := new(big.Int).Mod(p.x, P), new(big.Int).Mod(p.y, P)
+	x2, y2 := new(big.Int).Mod(p2.x, P), new(big.Int).Mod(p2.y, P)
+	if x1.Cmp(x2) == 0 {
+		temp := new(big.Int).Sub(P, y2)
+		temp.Mod(temp, P)
+		if y1.Cmp(temp) == 0 {
+			p.x.SetInt64(0)
+			p.y.SetInt64(0)
```

Comment:
I like the `point.Set` abstraction. In this particular case, would we need something like `p.Set(&point{new(big.Int), new(big.Int)})`? I don't think we can use `new` here since `*big.Int`s are pointers and will have a `nil` zero value.

Reply:
Ah. Indeed. I also have a thought about having a default constructor that does that i.e. `NewPoint()` which returns `&point{new(big.Int), new(big.Int)}` and then leave the assignment to the `Set` method so that line would be `p.Set(NewPoint())`. What do you think about that? Or even just `Inf()` so that it becomes `p.Set(Inf())`.

Reply:
I really like the `Inf` idea. I'll implement that.

Reply:
I just did.